### PR TITLE
Group views and member permissions

### PR DESCRIPTION
**Pull request by jwest60**
#75 #118 

- Added a form to allow the group owner to edit user permissions on the group page
- Enforced these permissions across all resources (i.e. users cannot view or edit on resources they do not have permission for)

image

### CODE REVIEW

**Comment by jwest60** [comment]
Finished writing tests, this PR is ready for review

File: app/Policies/MeasurementPolicy.php [hunk 1]
```diff
@@ -31,18 +31,6 @@ public function viewAny(User $user)
...
-
-        if($user->isFriendsWith($measurement->plant->user)) {
-            return true;
-        }
-
-        if ($user->isFriendsWith($measurement->plot->user)) {
-            return true;
-        }
-
-        if ($user->isFriendsWith($measurement->site->user)) {
```
Comment (almasaeed2010) [inline app/Policies/MeasurementPolicy.php:45]: Why were these removed? We still need to allow users to access a measurement. I think the best way to do this us to return `$user->can('view', $measurement->site)`

File: app/Policies/SitePolicy.php [hunk 2]
```diff
@@ -34,7 +34,17 @@ public function view(User $user, Site $site)
             return true;
         }
 
-        if($user->isFriendsWith($site->user)) {
+        if ($site->groups()->whereIn('group_id', $user->groups()
+            ->wherePivot('can_view', 1)->get()
+            ->pluck('pivot.group_id'))->exists()) {
```
Comment (almasaeed2010) [inline app/Policies/SitePolicy.php:39]: Is there a problem with simply plucking `id` here?
```suggestion
            ->pluck('id'))->exists()) {
```

File: resources/js/forms/UserForm.vue [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                        </label>
+                    </div>
+                    <div class="form-check">
+                        <input class="form-check-input"
+                               type="checkbox"
+                               value="1"
+                               :checked="form.can_edit"
+                               @change="form.can_edit = $event.target.checked"
+                               id="can-edit">
+                        <label class="form-check-label font-weight-normal" for="can-view">
```
Comment (almasaeed2010) [inline resources/js/forms/UserForm.vue:30]:
```suggestion
                        <label class="form-check-label font-weight-normal" for="can-edit">
```

File: app/Policies/SitePolicy.php [hunk 4]
```diff
@@ -34,7 +34,17 @@ public function view(User $user, Site $site)
...
+
+            return true;
+        }
+
+        if ($site->groups()->whereIn('id', $user->groups()
```
Comment (almasaeed2010) [inline app/Policies/SitePolicy.php:44]:
```suggestion
        if ($site->groups()->whereIn('groups.id', $user->groups()
```

File: app/Policies/SitePolicy.php [hunk 5]
```diff
@@ -61,6 +71,20 @@ public function create(User $user)
      */
     public function update(User $user, Site $site)
     {
+        if ($site->groups()->whereIn('group_id', $user->groups()
```
Comment (almasaeed2010) [inline app/Policies/SitePolicy.php:74]:
```suggestion
        if ($site->groups()->whereIn('groups.id', $user->groups()
```

File: app/Policies/SitePolicy.php [hunk 6]
```diff
@@ -61,6 +71,20 @@ public function create(User $user)
...
+            ->wherePivot('can_edit', 1)->get()
+            ->pluck('id'))->exists()) {
+
+            return true;
+        }
+
+        if ($site->groups()->whereIn('id', $user->groups()
```
Comment (almasaeed2010) [inline app/Policies/SitePolicy.php:81]:
```suggestion
        if ($site->groups()->whereIn('groups.id', $user->groups()
```

File: app/Policies/SitePolicy.php [hunk 7]
```diff
@@ -34,7 +34,17 @@ public function view(User $user, Site $site)
             return true;
         }
 
-        if($user->isFriendsWith($site->user)) {
+        if ($site->groups()->whereIn('group_id', $user->groups()
```
Comment (almasaeed2010) [inline app/Policies/SitePolicy.php:37]:
```suggestion
        if ($site->groups()->whereIn('groups.id', $user->groups()
```
